src/app/screens/gamepage/index.ts

```diff
@@ -76,7 +78,16 @@ export class GamePage {
     })
   }
 
+  private handleProgressBarChange() {
+    setInterval(() => {
+      const timeToNextWave = this.game.getNextAttack()
+      this.progressBar.style.width = ( timeToNextWave *100 /30).toString() + "%";
```

Comment:
const width = timeToNextWave * 100 / 30
this.progressBar.style.width = `${width}%`

Should work firn